aws-globalaccelerator-endpointgroup/src/test/kotlin/software/amazon/globalaccelerator/endpointgroup/ListHandlerTest.kt

```diff
@@ -0,0 +1,180 @@
+        assertEquals(1, response.resourceModels[0].portOverrides.size)
+        assertEquals(portOverrides1[0].listenerPort, response.resourceModels[0].portOverrides[0].listenerPort)
+        assertEquals(portOverrides1[0].endpointPort, response.resourceModels[0].portOverrides[0].endpointPort)
+        assertEquals(listenerArn, response.resourceModels[1].listenerArn)
+        assertEquals(model2.endpointGroupArn, response.resourceModels[1].endpointGroupArn)
+        assertEquals(model2.endpointGroupRegion, response.resourceModels[1].endpointGroupRegion)
+        assertEquals(1, response.resourceModels[1].portOverrides.size)
+        assertEquals(portOverrides2[0].listenerPort, response.resourceModels[1].portOverrides[0].listenerPort)
+        assertEquals(portOverrides2[0].endpointPort, response.resourceModels[1].portOverrides[0].endpointPort)
+    }
+
+    @Test
+    fun handleRequest_noEndpointGroups() {
+        val model = createTestResourceModel(endpointGroupArn1, endpointId1_1, endpointId1_2, portOverrides1)
+        val listEndpointGroupsResult = ListEndpointGroupsResult().withEndpointGroups(emptyList())
+        every { proxy.injectCredentialsAndInvoke(ofType(), ofType<ProxyListEndpointGroups>()) } returns listEndpointGroupsResult
+        val request = ResourceHandlerRequest.builder<ResourceModel>().desiredResourceState(model).build()
+        val response = ListHandler().handleRequest(proxy, request, null, logger)
+        assertNotNull(response)
+        assertEquals(OperationStatus.SUCCESS, response.status)
+        assertNotNull(response.resourceModels)
+        assertEquals(0, response.resourceModels.size)
+        assertNull(response.nextToken)
+    }
+
+    @Test
+    fun handleRequest_returnsMappedEndpointGroupsWithToken() {
+        val expectedNextToken = "This_token_is_expected"
+        val model = createTestResourceModel(endpointGroupArn1, endpointId1_1, endpointId1_2, portOverrides1)
+        val returnedPortOverrides = com.amazonaws.services.globalaccelerator.model.PortOverride()
+                .withListenerPort(portOverrides1[0].listenerPort)
+                .withEndpointPort(portOverrides1[0].endpointPort)
+        val endpointGroups = mutableListOf(
+                EndpointGroup()
+                        .withEndpointGroupArn(endpointGroupArn1)
+                        .withEndpointGroupRegion(endpointGroupRegion)
+                        .withTrafficDialPercentage(50.0f)
+                        .withEndpointDescriptions(createEndpointDescription(endpointId1_1, endpointId1_2))
+                        .withPortOverrides(returnedPortOverrides)
+        )
+        val listEndpointGroupsResult = ListEndpointGroupsResult()
+                .withEndpointGroups(endpointGroups)
+                .withNextToken(expectedNextToken)
+        every { proxy.injectCredentialsAndInvoke(ofType(), ofType<ProxyListEndpointGroups>()) } returns listEndpointGroupsResult
```

Comment:
same here, I think we should care that the token is right as a param provided in the request